homeassistant/components/solarlog/sensor.py

```diff
@@ -39,4 +39,9 @@ def __init__(
     @property
     def native_value(self) -> StateType:
         """Return the native sensor value."""
-        return self.coordinator.data[self.entity_description.json_key]
+        return (
+            getattr(self.coordinator.data, self.entity_description.key)
```

Comment:
Please store the result of getattr in a local variable and then conditionally apply the factor.

Reply:
Done